tests/e2e/idps_test.go

```diff
@@ -763,6 +761,193 @@ var _ = Describe("TF Test", func() {
+						)
+
+						By("Create gitlab idp using OCM api with the same parameters")
+						requestBody, _ := gitlabIdpOcmAPI.Type("GitlabIdentityProvider").
+							Name(gitLabIdpName).
+							Gitlab(cmv1.NewGitlabIdentityProvider().
+								ClientID(gitlabIDPClientId).
+								ClientSecret(gitlabIDPClientSecret).
+								URL(con.GitLabURL)).
+							MappingMethod("claim").
+							Build()
+						res, err := cms.CreateClusterIDP(ci.RHCSConnection, clusterID, requestBody)
+						Expect(err).ToNot(HaveOccurred())
+						Expect(res.Status()).To(Equal(http.StatusCreated))
+
+						By("Re-apply gitlab idp using tf manifests with same ocm api args")
+						err = idpService.gitlab.Apply(idpParam, false)
+						Expect(err).To(HaveOccurred())
+						Expect(err.Error()).Should(ContainSubstring(
+							"Identity Provider Name\n%s already exists", gitLabIdpName),
+						)
+
+						By("Delete gitlab idp from existing cluster")
+						defer cms.DeleteIDP(ci.RHCSConnection, clusterID, res.Body().ID())
+
+					})
+			})
+			Context("Author:smiron-Medium-OCP-65814 @OCP-65814 @smiron", func() {
+				It("OCP-65814 - IDP reconciliation - restore to an updated IDP config",
+					ci.Day2, ci.Medium, ci.FeatureIDP, func() {
+
+						gitLabIdpName = "OCP-65814-gitlab-idp-reconcil"
+
+						By("Create gitlab idp for existing cluster")
+						idpParam := &exe.IDPArgs{
+							ClusterID:    clusterID,
+							Name:         gitLabIdpName,
+							ClientID:     gitlabIDPClientId,
+							ClientSecret: gitlabIDPClientSecret,
+							URL:          con.GitLabURL,
+						}
+						Expect(idpService.gitlab.Apply(idpParam, false)).To(Succeed())
+						idpID, _ := idpService.gitlab.Output()
+
+						By("Edit gitlab idp using ocm api")
+
+						newClientSecret := h.GenerateRandomStringWithSymbols(30)
+						requestBody, _ := gitlabIdpOcmAPI.Type("GitlabIdentityProvider").
+							Gitlab(cmv1.NewGitlabIdentityProvider().
+								ClientID(gitlabIDPClientId).
+								ClientSecret(newClientSecret).
+								URL(con.GitLabURL)).
+							MappingMethod("claim").
+							Build()
+
+						resp, err := cms.PatchIDP(ci.RHCSConnection, clusterID, idpID.ID, requestBody)
+						Expect(err).ToNot(HaveOccurred())
+						Expect(resp.Status()).To(Equal(http.StatusOK))
+
+						// update is currently not supported for idp :: OCM-4622
+						By("Update and apply idp using terraform")
+						newClientSecret = h.GenerateRandomStringWithSymbols(30)
+						idpParam = &exe.IDPArgs{
+							ClusterID:    clusterID,
+							Name:         gitLabIdpName,
+							ClientID:     gitlabIDPClientId,
+							ClientSecret: newClientSecret,
+							URL:          con.GitLabURL,
+						}
+
+						err = idpService.gitlab.Apply(idpParam, true)
+						Expect(err).To(HaveOccurred())
+						Expect(err.Error()).Should(ContainSubstring(
+							"This RHCS provider version does not support updating an existing IDP"),
+						)
+
+						By("Delete gitlab idp using OCM API")
```

Comment:
Put this part the where the idp is created and make it in `defer`

Reply:
Is it meaningfull - verify idp deletion by API @xueli181114 ?

Reply:
fixed